### PR TITLE
Document PHP 8.3 support

### PR DESCRIPTION
**Pull request by come-nc**
### ☑️ Resolves

* Document https://github.com/nextcloud/server/pull/40630

### 🖼️ Screenshots

Screenshot_20231012_111338

Screenshot_20231012_111729

Screenshot_20231012_111825

### CODE REVIEW

**Review by kesselb — APPROVED** [review]

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_28.rst [hunk 1]
```diff
@@ -54,6 +54,11 @@ Removed APIs
 Back-end changes
 ----------------
 
+PHP 8.3
+^^^^^^^
+
+In this release support for PHP 8.3 was added. See release notes from PHP about new deprecations.
```
Comment (ChristophWurst) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_28.rst:60]: Nitpick: https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_24.html#php-8-1 was a bit more elaborative. Do we want to copy and adapt those instructions?

Comment (come-nc) [reply developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_28.rst:60]: I copied from 26 8.2 :confused:

Comment (come-nc) [reply developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_28.rst]: Done

**Review by ChristophWurst — APPROVED** [review]
:sunglasses:

**Pull request merged**